tests/variant_test.cpp

```diff
@@ -182,7 +182,7 @@ BOOST_AUTO_TEST_CASE( nested_objects_test )
 
       from_variant( v, sv1, nested_levels + 2 );
 
-      BOOST_CHECK( sv == sv1 );
+      BOOST_CHECK( decltype(sv)::type_eq()(sv, sv1) );
```

Comment:
I think it's expected that the two be totally equal, but not only `type_eq`.

Reply:
I hate to see code with an obvious error being merged. To avoid somewhere else abusing the `operator==` but still fix the code here, perhaps we can write a function E.G. `value_eq` or `full_eq`?

Reply:
IMO take the pragmatic approach for now and add
`BOOST_CHECK( sv.as<fc::test::item>() == sv1.as<fc::test::item>() );`